libs/novui/src/json-schema-components/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { JsonSchemaFormSectionVariant } from '../../styled-system/recipes';
+
+export const JSON_SCHEMA_FORM_ID_DELIMITER = '~~~';
+
+/** determine the tree depth of a JsonSchemaForm section with the given sectionId */
+export function calculateSectionDepth({ sectionId }: { sectionId: string }): number {
+  return sectionId.split(JSON_SCHEMA_FORM_ID_DELIMITER).length - 1;
+}
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** ‏Very neat!